src/System.Net.Security/tests/FunctionalTests/NegotiateStreamStreamToStreamTest.cs

```diff
@@ -214,7 +215,9 @@ public abstract class NegotiateStreamStreamToStreamTest
                 Assert.True(_sampleMsg.SequenceEqual(recvBuf));
 
                 client.Write(_sampleMsg, 0, _sampleMsg.Length);
-                server.Read(recvBuf, 0, _sampleMsg.Length);
+                // Test partial sync read.
+                server.Read(recvBuf, 0, PartialBytesToRead);
+                server.Read(recvBuf, PartialBytesToRead, _sampleMsg.Length - PartialBytesToRead);
```

Comment:
It'd be good to also validate the return value from Read.

Note that technically, these Read calls need not read the full amount requested.  It's possible that NegotiateStream's implementation with this test makes it so that this will always happen.

Reply:
I was confused about this comment, but now I understand it. :smile: 

> It's possible that NegotiateStream's implementation with this test makes it so that this will always happen.

That's true for byte count < `NegoState.MaxWriteDataSize` (that's why I opened the issue: NegotiateStream don't read full data, I thought it's a bug at first).

Will validate the return value from Read.